_posts/2024-07-19-Using-Redis-with-Sidecar.md

```diff
@@ -0,0 +1,179 @@
+
+        // Cache Time to Live, from Current Time to <specified> mins, after that cache will expire
+        cachingOption.AbsoluteExpirationRelativeToNow = absoluteExpiry ?? TimeSpan.FromSeconds(60);
+
+        // Cache Time to Live, will slide if it is accessed within the sliding window
+        cachingOption.SlidingExpiration = slidingExpiry;
+
+        // Convert to JSON
+        var dataToStore = JsonSerializer.Serialize(value);
+
+        // Set Cache
+        await cache.SetStringAsync(key, dataToStore, cachingOption);
+    }
+    ```
+
+5. **Running Your Application**
+Now you can run your application. Open your browser and navigate to:
+`http://localhost:5000/weatherforecast`.
+
+The browser will display the weather forecast in JSON format. Since we have implemented logging in the application, you will see logs in the command prompt indicating whether the response is coming from the database or the cache.
+
+## Deploying the Application to Linux App Service
+
+To get started, you'll need to containerize your .NET application. This [tutorial](https://learn.microsoft.com/en-us/dotnet/core/docker/build-container?tabs=windows&pivots=dotnet-8-0) walks you through the process step by step.
+*Note: The Sidecar feature is currently enabled for custom-container scenarios for Linux App Service. We are working on enabling it for code scenarios as well. We will update the blog soon for the code-based web applications*
+
+1. **Containerize Your .NET Application**
+    Create a Dockerfile in the root directory of your project. The Dockerfile for this project is [here](https://github.com/Azure-Samples/sidecar-samples/blob/main/Redis.Core.WebApi-master/Redis.Core.WebApi/Dockerfile).
+2. **Build and Push the Docker Image**
+    Build the image and push it to your preferred container registry, such as Azure Container Registry, Docker Hub, or a private registry.
+
+    ```bash
+    docker build -t <your-registry>/<your-image-name>:<tag> .
+    docker push <your-registry>/<your-image-name>:<tag>
+    ```
+
+    We would also recommend that you push the local redis image to ACR. This would allow you to take advantage of securing your resources using Managed Identity. Please refer to this [documentation](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli).
+3. **Create a New Linux Web App in Azure**
+    Create a new Linux Web App from the portal and choose the options for Container and Linux.
+    ![Create web app]({{site.baseurl}}/media/2024/07/CreateWebApp.jpg)
+
+    On the Container tab, make sure that Sidecar support is Enabled.
+
+    Specify the details of your application image.
+    ![Create web app]({{site.baseurl}}/media/2024/07/AddContainer.jpg)
+
+    *Note: We strongly recommend enabling [Managed Identity](https://learn.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=portal%2Chttp) for your Azure resources.*
```

Comment:
remove */en-us/* from links this help route folks to localized content

Reply:
Done.